### PR TITLE
ci: update release version for publish tag/release

### PR DESCRIPTION
**Pull request by gbockus-sf**
### What does this PR do?
When publishing with GHA for the first time we failed to create the tag/release in github due to how the version for the release was being read.  This PR fixes that access.  

### What issues does this PR fix or reference?
@W-11596486@

### Functionality Before
tag and release failed after publish to the marketplace. See https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/3895327384

### Functionality After
Tag and release are created correctly.  Verified in fork.  See https://github.com/gbockus-sf/salesforcedx-vscode/actions/runs/3896424448. Note the release failed in the fork but had the correct version. It was due to how I cobbled together the buildall step.

### CODE REVIEW

File: .github/workflows/publishVSCode.yml [hunk 1]
```diff
@@ -90,11 +92,13 @@ jobs:
         with:
           email: ${{ secrets.IDEE_GH_EMAIL }}
       - run: |
-            git tag v${{ inputs.version }}
-            git push origin v${{ inputs.version }}
+            git tag v${{ env.RELEASE_VERSION }}
+            git push origin v${{ env.RELEASE_VERSION }}
```
Comment (gbockus-sf) [inline .github/workflows/publishVSCode.yml:96]: run can't read input...runs can't read inputs...I'm burning it in

File: .github/workflows/publishVSCode.yml [hunk 2]
```diff
@@ -45,7 +45,7 @@ jobs:
       - run: rm ./SHA256
       - run: |
               git add SHA256.md
-              git commit -m "chore: updated SHA256 v${{ inputs.version }}"
+              git commit -m "chore: updated SHA256 v${{ inputs.version }} [skip ci]"
```
Comment (randi274) [inline .github/workflows/publishVSCode.yml:48]: But can run read inputs?!

File: .github/workflows/publishOnMergeWIthMain.yml [hunk 3]
```diff
@@ -3,7 +3,6 @@ on:
   push:
     branches: 
       - 'main'
-  workflow_dispatch:
```
[the remark this answers is not in the document]
Comment (gbockus-sf) [reply .github/workflows/publishOnMergeWIthMain.yml:6]: yeah it didn't have one originally...and it was added b/c the syntax was off originally when a push to main happens.  I added it on the fly to get us releasing and removed it as a cleanup step

File: .github/workflows/publishVSCode.yml (hunk 2 above)
Comment (gbockus-sf) [reply .github/workflows/publishVSCode.yml:48]: ha good catch...must not have burned it yet

**Review by randi274 — APPROVED** [review]

**Pull request merged**